src/v/cloud_storage/remote_partition.cc

```diff
@@ -483,10 +483,11 @@ class partition_record_batch_reader_impl final
     }
 
     ss::future<> init_cursor(storage::log_reader_config config) {
-        auto segment_unit
-          = co_await _partition->materialized().get_segment_units();
+        auto segment_unit = co_await _partition->materialized()
+                              .get_segment_units(config.abort_source);
```

Comment:
If we can be certain that the lifetime of a partition_record_batch_reader_impl is strictly contained by the lifetime of the kafka connection, then we should be able to store this abort source ref. as a member during `start` instead of always copying it from the nearest log reader config. But this lifetime relation is hard to assert.

Reply:
I think it's guaranteed. The partition reader is created per fetch request. It's created, used, and disposed in the `read_from_partition` call in `fetch.cc`. The segment reader stored inside the partition reader can outlive the fetch request.

Reply:
Since at all times when waiting for units in a partition reader we need to watch out for the abort source from the connection, and given that the connection outlives the reader, in the follow up PR I will change the abort source ref. to be stored in the partition reader.